arch/risc-v/src/common/riscv_percpu.c

```diff
@@ -86,13 +92,39 @@ void riscv_percpu_init(void)
 
 void riscv_percpu_add_hart(uintptr_t hartid)
 {
-  /* Hart IDs go from 0...4 */
+  riscv_percpu_t *percpu;
+  uintptr_t       stack_top;
+
+  /* Get free entry for this hart, this must not fail */
+
+  percpu = (riscv_percpu_t *)sq_remfirst(&g_freelist);
+  DEBUGASSERT(percpu);
+
+  /* Get interrupt stack (if any) */
+
+  stack_top = 0;
```

Comment:
I mean remove line 105, your code at line 111 and 116 assign stack_top again. Why need assign stack_top to zero here?

Reply:
Both assignments are dependent on this:
`#if CONFIG_ARCH_INTERRUPTSTACK > 15`

Which means this:
`percpu->irq_stack = stack_top;`

Would use an uninitialized value.

I can of course remove the temp variable altogether, but to me this current implementation is fine too (need to fix the bug below).

Reply:
Sorry, I miss the CONFIG_ARCH_INTERRUPTSTACK check, but should we remove irq_stack field in this case?

Reply:
Please, check the new patch. The old patch was just wrong. Using hartid does not work for calculating stack position.

It is a fully valid configuration to run NuttX with 2 cores and select harts 3 and 4. This will obviously fail.

Reply:
Resolving as obsolete